src/platform/forms-system/src/js/fields/ObjectField.jsx

```diff
@@ -290,7 +291,8 @@ class ObjectField extends React.Component {
       </div>
     );
 
-    if (title) {
+    // fieldset wasn't injected in FieldTemplate so add it here if there is a title specified.
+    if (title && !useFieldsetLegend) {
```

Comment:
ObjectField has logic above for when it does or doesn't show a title. Instead of injecting this `useFieldsetLegend` prop, could we use that logic?

Reply:
I think that's a better idea. Let me take a look 👍

Reply:
Hmm actually I think going off the above title logic might not be feasible, since the `fieldset` here wraps the entirety of `fieldcontent`, not just whatever title is rendered. But, it still might be doable to recreate some similar logic to `useFieldsetLegend` rather than injecting it as a prop.

Reply:
Yeah, that's more of what I meant, recreating it. The current `if (title)` check seems pretty pointless and since there's similar logic to what's in FieldTemplate for rendering the title already, injecting the prop doesn't feel like it consolidates the logic in one place, which to me would be the main reason for considering injecting a prop like that.

Reply:
I've updated the check to:
```
    const renderFieldset = !!showFieldLabel;
    if (title && !renderFieldset) {
      return <fieldset className={fieldsetClassNames}>{fieldContent}</fieldset>;
    }
```

I think `title` still needs to be included here, in the event uiSchema is using an empty object with `ui:description` to display a chunk of helper text. Otherwise, `fieldset` will wrap that block and cause the same accessibility error for a missing legend.